docs/src/contributing/git-workflow.md

```diff
@@ -51,36 +101,147 @@ If the branch exists in a triton/neptune official repo, (as opposed to a persona
+exposed to peers running older versions of the software and may shake out issues
+that do not occur with automated testing.
+
+#### Some specifics
+
+1. The release candidate should be published using the same release process as
+   the eventual release; see [releasing](releasing.md).
+2. Each release candidate must be publicly announced with a dedicated topic in
+   [talk.neptune.cash](https://talk.neptune.cash).
+3. The release candidate should target 2 weeks of testing after the announcement
+   on average with a minimum of 1 week.
+4. The company behind neptune-cash will provide at least 1 dedicated machine for
+   the public testnet for the purpose of running the release-candidate binary
+   and composing blocks.  Of course the community is also encouraged to run
+   nodes and mine (compose or guess) if possible.
+5. A set of automated tests will be created that utilize the RPC API to perform
+   automated transactions and test basic functionality of the live running
+   nodes. These tests should/must be run at least once against each
+   release-candidate binary while connected to the public testnet network.
+6. If the release-candidate introduces breaking consensus changes, then a
+   reset/restart of the testnet from the genesis block may be performed.
+7. Patch releases may skip testnet testing if the risk seems low and/or the
+   urgency is sufficiently high.
+
+
+# neptune-core versioning and releases
+
+## deviation from standard semver practice
+
+neptune-core is presently both a (blockchain) library and a binary. Semantic versioning signals changes to APIs and mainly applies to libraries.  But it does not capture breaking changes that can occur in a blockchain ecosystem or breaking changes with regards to stored state, ie databases.
+
+For blockchains it is of primary importance to maintain consensus and network
+compatibility with other nodes on the network.
+
+For these reasons, we define the version identifier as follows.
+
+## version identifier
+
+neptune-core uses sem-ver compatible identifier of the form:
+```
+<major>.<minor>.<patch>[-<candidate>]
+
+where `candidate` is of the form "rc1", "rc2", etc and starts at 1.
+```
+
+Examples:
+```
+0.4.0-rc1
+0.5.0-rc1
+0.5.0-rc2
+0.5.0-rc3
+1.2.2
+```
+
+Non-Examples: (should not occur)
+```
+0.4.1-rc1       (point release should not be a release-candidate)
+1.2.0-rc0       (release candidates start at 1, not 0)
+```
+
+note: for git tags, a "v" is prefixed, as per git convention.
+
+Our release process treats these as:
+
+**major**: bumped any time there are:
+
+1. significant consensus or p2p layer changes. eg hard or soft-fork or breaking protocol change.
+2. breaking library API changes.
+3. breaking database changes.
+
+resets minor and patch to 0.
+
+**minor**: bumped any time changes from master are included in a release candidate.
+
+resets patch to 0.
+
+**patch**: bumped any time changes from a release branch are included in a release.
+
+resets candidate to "-rc1".
```

Comment:
Based on the picture in the github workflow, a patch release resets the suffix to "" (empty string). Specifically, the version evolves from 0.4.0 to 0.4.1 without suffix. The suffix is only added when a new release candidate is made.

Reply:
thx, I'll revisit this.